DOCSP-47548 Updates the installation page to include deployment considerations (#217)
mongodb/docs-relational-migrator: source/installation.txt

and the recommended migration scenario for each deployment model:

.. list-table::
   :header-rows: 1
   :widths: 15 35 20 30

   * - Deployment Model

     - Description

     - Use Cases

     - Details

   * - Local Desktop

     - Install Relational Migrator to a single machine that cannot 
       be externally accessed.

       For more information, see :ref:`<local-machine-install>`.
      
     - 
       - Evaluation
       - Data Modeling
       - Testing

     - Evaluation and small production jobs expected to run for less than a day. Note 
       that larger migrations can be broken down into separate jobs using 
       :ref:`table filters <rm-table-filters>`.

   * - Unattended Server/:abbr:`VM (Virtual Machine)`

     - Install Relational Migrator on an unattended server that binds Relational 
       Migrator to an IP address and port and exposes it as a web application.
       
       For more information, see :ref:`<unattended-server>`.

     - 
       - Testing
       - Smaller (<100GB) Production Migrations

     - Suitable for the majority of use cases, including tests and small production migrations. 

>        Note that installing as an unattended server is not a highly-availably solution, and 

Reviewer: ```suggestion
       Note that installing as an unattended server is not a highly-available solution, and 
```

Author: Good catch, thank you!

   :depth: 1
   :class: singlecol

.. _rm-deployment-considerations:

Supported Deployment Models
---------------------------

> The following table describes {+rel-mig+}'s supported deployment models  
> and the recommended use cases for each deployment model:

Reviewer 2: [optional suggestion] if you want to[ limit the use of recommendations ](https://www.mongodb.com/docs/meta/style-guide/writing/write-to-the-user/#:~:text=%20Limit%20how%20often%20you%20state%20recommendations.%20)

```suggestion
The following table describes {+rel-mig+}'s supported deployment models  
and example use cases for each deployment model:
```

.. list-table::
   :header-rows: 1

   * - Deployment Model
     - Use Cases

   * - :ref:`<rm-install-scenario-desktop>`
     - 
       - Evaluation
       - Data Modeling
       - Testing
       - Small Production Migrations (Under 100GB)

   * - :ref:`<rm-install-scenario-server>`
     - 
       - Testing
       - Production Migrations (Under 1TB)

   * - :ref:`<rm-install-scenario-kafka>`
     - 
       - Mission Critical Migrations
       - Long Running Migrations

   * - :ref:`<rm-install-scenario-confluent>` 
     - 
       - Mission Critical Migrations
       - Long Running Migrations

.. _rm-install-scenario-desktop:

Local Desktop
~~~~~~~~~~~~~

You can install {+rel-mig+} to a single machine that cannot be externally accessed.

> Installing locally is suitable for development, evaluation, and small production jobs 
> that are expected to run for less than one day.

Reviewer 2: [suggestion] for clarity

```suggestion
Local installations are suitable for development, evaluation, and small production jobs 
that are expected to run for less than one day.
```

.. note:: 
  

>    If you must perform a larger migration, you can split the migration into multiple 
>    jobs using :ref:`table filters <rm-table-filters>`.

Reviewer 2: [suggestion] for clarity

```suggestion
   If you must perform a large migration, you can split the migration into multiple 
   jobs by using :ref:`table filters <rm-table-filters>`.
```

For more information, see :ref:`<local-machine-install>`.

.. _rm-install-scenario-server:

Unattended Server/VM
~~~~~~~~~~~~~~~~~~~~

You can install {+rel-mig+} on an unattended server or :abbr:`VM (Virtual Machine)` 
that binds {+rel-mig+} to an IP address and port, exposing it as a web application.
Installing {+rel-mig+} on an unattended server is suitable for the majority of use cases, 
including tests and production migrations.

.. note::
   
   Installing {+rel-mig+} on an unattended server is not a highly-available solution. 

>    Users must monitor the server and intervene if an application issue occurs. 

Reviewer 2: [suggestion] to start with the [conditional clause](https://www.mongodb.com/docs/meta/style-guide/writing/write-clear-concise-sentences-paragraphs/#:~:text=Sentences%20that%20describe%20a%20condition%20should%20always%20begin%20with%20theconditional%20clause)

```suggestion
   If an application issue occurs, users must monitor the server and intervene.
```

Author: Updated - I removed the "monitoring" portion as that part is not conditional. I think it is probably okay for it to be removed, but will double check with eng and add a separate sentence related to monitoring if necessary.

For more information, see :ref:`<unattended-server>`.

.. _rm-install-scenario-kafka:

Kafka Cluster
~~~~~~~~~~~~~

`Apache Kafka <https://kafka.apache.org/>`__ is an open-source platform for distributed 

> workloads. If you are using {+rel-mig+} for critical production workloads or long-running 

Reviewer 2: [suggestion] for clarity

```suggestion
workloads. If you use {+rel-mig+} for critical production workloads or long-running 
```

> :abbr:`CDC (Change Data Capture)` syncs, Kafka is recommended. Using {+rel-mig+}

Reviewer 2: [suggestion] for active voice

```suggestion
:abbr:`CDC (Change Data Capture)` syncs, MongoDB recommends Kafka. Using {+rel-mig+}
```

     - Description

     - Recommended scenario

   * - Local

     - Install Relational Migrator to a single machine that cannot 
       be externally accessed.

       For more information, see :ref:`<local-machine-install>`.

     - Evaluation and proof of concept jobs that run for less than four hours.
    
   * - Docker

     - Install Relational Migrator in a way that is simple to replicate
       and independent of the operating system.

       For more information, see :ref:`<rm-install-docker>`.

     - Evaluation and proof of concept jobs that run for less than four hours,
       and do not run directly on a server or machine.
       

   * - Unattended server

     - Install Relational Migrator on an unattended server that binds Relational 

>        Migrator to an IP address and exposes it as a web application.

Reviewer 3: [question]

Does it bind to a specific port as well? If so, consider mentioning this.

       
       For more information, see :ref:`<unattended-server>`.

     - Small production jobs with runtimes between four hours and four weeks,
       or involve a database smaller than 20GB.
      

   * - Kafka

     - `Apache Kafka <https://kafka.apache.org/>`__ is an open-source platform for distributed 
       workloads.

       Use Relational Migrator as a Kafka Connect plugin to improve resilience and scalibility
       of big data migration jobs.

       For more information, see :ref:`kafka-intro`.

     - Production data migrations that run longer than four weeks, or involve databases larger than 20GB.

       

   * - Kafka with Confluent Cloud

     - `Confluent  <https://www.confluent.io/>`__ is a fully-managed Kafka hosting 
       provider run by the founders of Kafka. 

       Use APIs to add the newest Relational Migrator sink connectors to Confluent Cloud
       as managed connectors.

       For more information, see :ref:`install-confluent-kafka`.

     - Production jobs with the reliability of Kafka without having to manage your own cluster.

> Deployment Considerations

Reviewer 3: [suggestion]

For this **Deployment Considerations** section, I think we should create includes from the existing copy so that we don't have to maintain duplicate copy later.

Author: Good call - I'm sure future me will appreciate this suggestion when I inevitably need to reuse some of these 😅 

I broke them up into includes at each sub-heading
